paddle/fluid/framework/operator.cc

```diff
@@ -1099,6 +1102,19 @@ Scope* OperatorWithKernel::PrepareData(
       auto* trans_var = new_scope->Var(var_name);
       input_vars[i] = trans_var;
 
+      if (is_inplace) {
+        for (auto& out_var_name_item : Outputs()) {
+          std::vector<Variable*>& output_vars =
+              ctx->outputs[out_var_name_item.first];
+          for (size_t i = 0; i < out_var_name_item.second.size(); ++i) {
+            auto& out_var_name = out_var_name_item.second[i];
+            if (out_var_name == var_name) {
+              output_vars[i] = input_vars[i];
+            }
+          }
+        }
+      }
+
       Tensor out;
       TransformData(expected_kernel_key, kernel_type_for_var, *tensor_in, &out);
       SetTensorToVariable(*var, out, trans_var);
```

Comment:
@xiaolil1 Could you please explain why scenario you are describing : non-mkldnn tensor --> mkldnn tensor  when op is doing inplace computation does not work without this PR? My understanding is that
if this is inplace then var_name for input and output is the same. Both output and input var_names should point to the same variable , so by calling SetTensorToVariable we should have both input and output updated?

Reply:
Hi @jczaja , thanks for your quickly review. As you said, var_name for input and output is the same and point to the same tensor at the beginning, but "data transform" between "non-mkldnn tensor --> mkldnn" tensor will create new tensor and replace the original input tensor, you can find it in line:1100 ( i have marked it), while output doesn't. So after the replacement, input and output will not point to same tensor, we need to add the same mechanism as input to keep the "in-place". What's more, i think not only the "non-mkldnn tensor --> mkldnn", but also other cases need "data_transform" will meet this issue. @hshen14